gitpod-io/gitpod
# [content-service] Fix backup restore

## Description

Partial revert of https://github.com/gitpod-io/gitpod/pull/9589. Only reverts the **restore** part.

## How to test
- Open a workspace. In the terminal execute:
```
touch basic
sudo touch root-file
touch executable-file
chmod +x executable-file
touch suid-file
chmod 2555 suid-file
mkdir suid-directory
chmod g+s suid-directory
```
- Check the output of `ls -lat`, should be similar to
```
drwxr-x--- 8 gitpod gitpod 4096 Apr 28 17:31 .
-r-xr-sr-x 1 gitpod gitpod    0 Apr 28 17:31 suid-file
drwxr-sr-x 2 gitpod gitpod    6 Apr 28 17:31 suid-directory
-rwxr-xr-x 1 gitpod gitpod    0 Apr 28 17:30 executable-file
-rw-r--r-- 1 root   root      0 Apr 28 17:30 root-file
-rw-r--r-- 1 gitpod gitpod    0 Apr 28 17:30 basic
```
- Stop and open the workspace
- Open a terminal and run `ls -lat`
- Compare the outputs

## Release Notes

```release-note
[content-service] Fix backup restoration
```

xref https://github.com/gitpod-io/gitpod/pull/9730
